### ISSUE TITLE
Dependency Guava 18.0 vulnerability CVE-2018-10237

### ISSUE THREAD

**Issue by marco-schmidt** [eclipse/rdf4j#2473]
Upgrading to rdf4j 3.4.0 made my [project am's build](https://github.com/marco-schmidt/am) fail due to a vulnerability in dependency Guava 18.0:
```
 >./gradlew check
Dependency verification is an incubating feature.

> Task :dependencyCheckAnalyze
Verifying dependencies for project am
Checking for updates and analyzing dependencies for vulnerabilities
Generating report for project am
Found 1 vulnerabilities in project am

One or more dependencies were identified with known vulnerabilities in am:

guava-18.0.jar (pkg:maven/com.google.guava/guava@18.0, cpe:2.3:a:google:guava:18.0:*:*:*:*:*:*:*) : CVE-2018-10237

...

BUILD FAILED in 5m 23s
10 actionable tasks: 8 executed, 2 up-to-date
```
 https://travis-ci.org/github/marco-schmidt/am/jobs/720889106#L1436

And apparently I'm getting this as a transitive dependency from rdf4j:
```
./gradlew dependencies
+--- org.eclipse.rdf4j:rdf4j-repository-sparql:3.4.0
|    +--- org.eclipse.rdf4j:rdf4j-repository-api:3.4.0
|    |    +--- org.eclipse.rdf4j:rdf4j-query:3.4.0
|    |    |    +--- org.eclipse.rdf4j:rdf4j-model:3.4.0
|    |    |    |    +--- org.eclipse.rdf4j:rdf4j-util:3.4.0
|    |    |    |    |    \--- org.slf4j:slf4j-api:1.7.30
|    |    |    |    \--- com.google.guava:guava:18.0
```

Since vulnerability CVE-2018-10237 is relatively old and Guava version 18 seems to have been in rdf4j for at least seven months: https://github.com/eclipse/rdf4j/blame/2619b87a50dd21f16a80343a125164dd5684f3c3/pom.xml#L309 I don't quite know why the alert was raised only now. I've fixed this by enforcing an upgrade to guava 29.0-jre in my own build: https://github.com/marco-schmidt/am/blob/488a126464003bd2a83b785d94ace0aadb39fb1c/build.gradle#L50

Maybe you want to upgrade that dependency as well? And maybe someone can help me understand why this wasn't an issue before.

Greetings,
Marco

**Comment by barthanssens** [comment]
Hi Marco,

thanks for the heads-up, it is related to (or a duplicate of ) #1097.

Last time I've checked, upgrading to a newer guava version did break some other parts of the tree (e.g. solr), but I haven't checked v29.0... so I'll have check again if everything compiles (and all compliance checks pass) with the version you mentioned.

**Comment by jeenbroekstra** [comment]
The reason this didn't flag up in your project before is probably that we've only started adding guava as a dependency in the rdf4j-model module since release 3.4 (as part of performance improvents related to #2442). Before, it was included in other modules (like for example the queryalgebra), but if  your project does not use those modules, it will not have been an issue for you in earlier releases.

**Comment by jeenbroekstra** [comment]
We'll fix as part of #1097

**Comment by marco-schmidt** [comment]
@barthanssens @jeenbroekstra Thanks for the explanations! I didn't find the old ticket searching for guava 18 for some reason. Thanks for maintaining rdf4j.

Greetings,
Marco
